server/cast/interval.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cast
+
+import (
+	"github.com/dolthub/go-mysql-server/sql"
+
+	"github.com/dolthub/doltgresql/postgres/parser/duration"
+
+	"github.com/dolthub/doltgresql/server/functions/framework"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// initInterval handles all casts that are built-in. This comprises only the "From" types.
+func initInterval() {
+	intervalAssignment()
+	intervalExplicit()
+	intervalImplicit()
+}
+
+// intervalAssignment registers all assignment casts. This comprises only the "From" types.
+func intervalAssignment() {
+	framework.MustAddAssignmentTypeCast(framework.TypeCast{
+		FromType: pgtypes.Interval,
+		ToType:   pgtypes.BpChar,
+		Function: func(ctx *sql.Context, val any, targetType pgtypes.DoltgresType) (any, error) {
+			return handleStringCast(val.(duration.Duration).String(), targetType)
+		},
+	})
```

Comment:
I don't see this cast (or the others) in Postgres. You can get all of the casts using:
```sql
SELECT 
	c.oid,
	src_typ.typname AS castsource,
	tgt_typ.typname AS casttarget,
	c.castfunc,
	c.castcontext,
	c.castmethod
FROM 
	pg_catalog.pg_cast c
JOIN 
	pg_catalog.pg_type src_typ ON c.castsource = src_typ.oid
JOIN 
	pg_catalog.pg_type tgt_typ ON c.casttarget = tgt_typ.oid
LEFT JOIN 
	pg_catalog.pg_proc p ON c.castfunc = p.oid
ORDER BY
	src_typ.typname, tgt_typ.typname;
```

Reply:
Thank you! 🙏